build.gradle

```diff
@@ -598,11 +598,43 @@ configure(updateBuildScript) {
 
 // Deobfuscation
 
-def deobf(String sourceURL, String fileName) {
-    def bon2File  = "$projectDir/deobf/BON2-2.5.0.jar"
-    def cacheFile = "$projectDir/deobf/out/" + fileName + ".jar"
-    def deobfFile = "$projectDir/deobf/out/" + fileName + "-deobf.jar"
+def deobf(String sourceURL) {
+    try {
+        URL url = new URL(sourceURL)
+        String fileName = url.getFile()
+
+        //get rid of directories:
+        int lastSlash = fileName.lastIndexOf("/")
+        if(lastSlash > 0) {
+            fileName = fileName.substring(lastSlash + 1)
+        }
+        //get rid of extension:
+        if(fileName.endsWith(".jar")) {
+            fileName = fileName.substring(0, fileName.lastIndexOf("."))
+        }
+
+        String hostName = url.getHost()
+        if(hostName.startsWith("www.")) {
+            hostName = hostName.substring(4)
+        }
+        List parts = Arrays.asList(hostName.split("\\."))
+        Collections.reverse(parts)
+        hostName = String.join(".", parts)
 
+        return deobf(sourceURL, hostName + "/" + fileName)
+    } catch(Exception e) {
+        return deobf(sourceURL, "deobf/" + String.valueOf(sourceURL.hashCode()))
+    }
+}
+
+// The method above is to be prefered. Use this method if the filename is not at the end of the URL.
+def deobf(String sourceURL, String fileName) {
+    String cacheDir = System.getProperty("user.home") + "/.gradle/caches/"
```

Comment:
This looks like it should work cross-platform, right?

Reply:
Yes, you can check it, when you tried to build ModTweaker earlier today, this change was already applied